inc/class.client.php

```diff
@@ -288,6 +288,25 @@ public static function get_rel_attribut() {
 		return $rel;
 	}
 
+	/**
+	 * Detect the texonomy based on the item or context
+	 * 
+	 * @param string $item
+	 * @return string|false The detected taxonomy or false if not detected
+	 */
+	public static function detect_taxonomy($item){
```

Comment:
This is not efficient as it'll cause large site to load very slow.

We can't call every taxonomy on the site in post loop then check each terms.

If you can't find more suitable method by changing current implementation, I think you should just remove taxonomy in `get_term_by` and limit it to `$term = get_term_by('name', $term_name);`

If you check the documentation, taxonomy parameter is optional https://developer.wordpress.org/reference/functions/get_term_by/